src/app/components/Navbar.tsx

```diff
@@ -62,29 +77,43 @@ const Navbar = () => {
           </div>
         </div>
         <div>
-          <div
-            className={`flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0 ${navbar ? 'block' : 'hidden'}`}
-          >
-            <ul className="items-center justify-center space-y-8 md:flex md:space-x-6 md:space-y-0">
-              {menu.map(({ name, url }, index) => {
-                const isActive = searchParams.get('tag') === url.query?.tag;
-                return (
-                  <li
-                    key={index}
-                    className={`block text-sm font-semibold text-gray-400 hover:bg-indigo-50 active:bg-indigo-100 hover:text-indigo-600 rounded ${
-                      isActive && 'text-indigo-50'
-                    }`}
-                  >
-                    <Link
-                      className="block w-full p-4"
-                      onClick={() => setNavbar(prev => (prev ? false : prev))}
-                      href={url}
-                    >
-                      {name}
-                    </Link>
-                  </li>
-                );
-              })}
+          <div className="md:hidden">
+            <div
+              tabIndex={-1}
+              id="drawer-navigation"
+              className={`fixed top-0 left-0 z-40 w-64 h-screen p-4 overflow-y-auto transition-transform bg-gray-800 shadow ${navbar ? 'transform-none' : '-translate-x-full'}`}
+              aria-labelledby="drawer-navigation-label"
+            >
+              <div className="py-4 overflow-y-auto">
+                <ul className="space-y-2 font-medium">
+                  <MenuLinks toggle={toggleNavbar} navbar={navbar} />
+                </ul>
+
+                <div className="border-t-[.5px] my-3 border-gray-300/20" />
+
+                <ul className="flex flex-wrap items-center justify-start">
+                  {otherTags.map((tag, ix) => {
+                    const isActive = tag === searchParams.get('tag');
+                    return (
+                      <li
+                        key={ix}
+                        className={`text-xs font-medium mx-1 my-1 text-gray-500 active:bg-indigo-100 hover:text-indigo-600 rounded ${isActive && 'text-indigo-100'}`}
+                      >
+                        <Link className="px-1" onClick={toggleNavbar} href={{ pathname: '/Duas', query: { tag: tag } }}>
+                          {tag}
+                        </Link>
+                      </li>
+                    );
+                  })}
+                </ul>
+              </div>
+            </div>
+            {navbar && <div className="fixed top-0 left-0 z-30 bg-black/10 w-full h-full" onClick={toggleNavbar}></div>}
+          </div>
+
+          <div className={`flex-1 justify-self-center pb-3 mt-8 md:block hidden md:pb-0 md:mt-0`}>
+            <ul className="items-center justify-center space-y-8 md:flex lg:space-x-6 md:space-x-2 md:space-y-0">
+              <MenuLinks toggle={toggleNavbar} />
```

Comment:
Can't we extract and have separate components? My main purpose is to divide it into small components and enhance readability. RN, our code has very low readability. 

Also, feel free to add some comments/notes to illustrate wherever something is special.